controller/network/router.go

```diff
@@ -179,3 +239,45 @@ func (ctrl *RouterController) UpdateCachedFingerprint(id, fingerprint string) {
 		}
 	}
 }
+
+type RouterLinks struct {
+	sync.Mutex
+	atomic.Value
+}
+
+func (self *RouterLinks) GetLinks() []*Link {
+	result := self.Load()
+	if result == nil {
+		return nil
+	}
+	return result.([]*Link)
+}
+
+func (self *RouterLinks) Add(link *Link) {
+	self.Lock()
+	defer self.Unlock()
+	links := self.GetLinks()
+	newLinks := make([]*Link, 0, len(links)+1)
+	for _, l := range links {
+		newLinks = append(newLinks, l)
+	}
+	newLinks = append(newLinks, link)
+	self.Store(newLinks)
+}
+
+func (self *RouterLinks) Remove(link *Link) {
```

Comment:
should these functions have metrics - or are they somewhere that's just not obvious?

Reply:
We have some events for these -> router online/offline, which is probably the most common reason links are removed. Don't have actual link events. Should probably talk to ops and see how useful those would be